crates/typst-cli/src/world.rs

```diff
@@ -339,6 +386,18 @@ fn read(path: &Path) -> FileResult<Vec<u8>> {
     }
 }
 
+/// Read from stdin.
+fn read_from_stdin() -> FileResult<Vec<u8>> {
+    let mut buf = Vec::new();
+    let result = io::stdin().read_to_end(&mut buf);
+    match result {
+        Ok(_) => (),
+        Err(e) if e.kind() == io::ErrorKind::BrokenPipe => (),
+        Err(err) => return Err(FileError::Other(Some(eco_format!("{err}")))),
```

Comment:
This should probably use `FileError::from_io` rather than always using the `Other` variant.